launcher.py

```diff
@@ -0,0 +1,358 @@
+    """
+
+    def __init__(self, name, package_name={}, check_cmd={}, install_cmd={}) -> None:
+        self.name = name
+        self.package_name = package_name
+        self.check_cmd = check_cmd
+        self.install_cmd = install_cmd
+
+    def check_dependency(self, system, install):
+        """
+        Checks if the dependency is installed. If not, installs it if `install` is True.
+
+        Args:
+            system (str): The operating system type.
+            install (bool): Whether to install the dependency if not found.
+        """
+        if system in self.check_cmd:
+            check_cmd = self.check_cmd[system]
+        elif "all" in self.check_cmd:
+            check_cmd = self.check_cmd["all"]
+        else:
+            check_cmd = [self.package_name.get(system, self.name) + " --version"]
+
+        try:
+            for cmd in check_cmd:
+                if type(cmd) == str:
+                    execute_command(cmd)
+                else:
+                    cmd()
+
+        except (FileNotFoundError, subprocess.CalledProcessError):
+            if not install:
+                print(
+                    f"{self.name} is not installed. Use --install to install dependencies automatically or install manually."
+                )
+                sys.exit(1)
+            self.install_dependency(system)
+
+    def install_dependency(self, system):
+        """
+        Installs the dependency for the specified operating system.
+
+        Args:
+            system (str): The operating system type.
+        """
+        if system in self.install_cmd:
+            install_cmd = self.install_cmd[system]
+        elif "all" in self.install_cmd:
+            install_cmd = self.install_cmd["all"]
+        else:
+            if system == "Linux":
+                import distro
+
+                linux_distro = distro.id().lower()
+                if linux_distro in ["ubuntu", "debian", "kali"]:
+                    install_cmd = [
+                        "sudo apt-get update",
+                        f"sudo apt-get install -y {self.package_name.get(system, self.name)}",
+                    ]
+                elif linux_distro in ["fedora", "centos", "rhel"]:
+                    install_cmd = [f"sudo dnf install -y {self.package_name.get(system, self.name)}"]
+                else:
+                    print(f"Unsupported Linux distribution. Please install {self.name} manually.")
+                    sys.exit(1)
+            elif system == "Darwin":
+                install_cmd = [f"brew install {self.package_name.get(system, self.name)}"]
+            elif system == "Windows":
+                if not check_msys2_installed():
+                    install_msys2()
+                install_cmd = [f"pacman -S --needed --noconfirm {self.package_name.get(system, self.name)}"]
+            else:
+                print(f"Unsupported operating system: {system}. Please install {self.name} manually.")
+                sys.exit(1)
```

Comment:
This is a little too much nesting and I believe we can do better. How about you introduce member functions for each platform so this reads much lighter?

Perhaps introduce an install() fn that will ask for the appropriate installer from a _get_installer(). The installers could be stored in a hash map. So you'd be injecting this dependency where you're intending to use it. It should eliminate the need for a lot of the control flow here as well. 

Btw, a side note, can you introduce a leading underscore for private member functions?